cmd/internal/cli/build_linux.go

```diff
@@ -103,169 +98,182 @@ func runBuild(cmd *cobra.Command, args []string) {
 		sylog.Fatalf("%s", err)
 	}
 
-	if remote {
-		// building encrypted containers on the remote builder is not currently supported
-		if encrypt {
-			sylog.Fatalf("Building encrypted container with the remote builder is not currently supported.")
-		}
+	if buildArgs.remote {
+		runBuildRemote(cmd, dest, spec)
+	} else {
+		runBuildLocal(cmd, dest, spec)
+	}
+	sylog.Infof("Build complete: %s", dest)
+}
 
-		handleRemoteBuildFlags(cmd)
+func runBuildRemote(cmd *cobra.Command, dst, spec string) {
+	// building encrypted containers on the remote builder is not currently supported
+	if buildArgs.encrypt {
+		sylog.Fatalf("Building encrypted container with the remote builder is not currently supported.")
+	}
 
-		// submitting a remote build requires a valid authToken
-		if authToken == "" {
-			sylog.Fatalf("Unable to submit build job: %v", remoteWarning)
-		}
+	handleRemoteBuildFlags(cmd)
 
-		def, err := definitionFromSpec(spec)
-		if err != nil {
-			sylog.Fatalf("Unable to build from %s: %v", spec, err)
-		}
+	// submitting a remote build requires a valid authToken
+	if authToken == "" {
+		sylog.Fatalf("Unable to submit build job: %v", remoteWarning)
+	}
 
-		if sandbox {
-			// create temporary file to download sif
-			f, err := ioutil.TempFile(tmpDir, "remote-build-")
-			if err != nil {
-				sylog.Fatalf("Could not create temporary directory: %s", err)
-			}
-			os.Remove(f.Name())
-			dest = f.Name()
-
-			// remove downloaded sif
-			defer os.Remove(f.Name())
-
-			// build from sif downloaded in tmp location
-			defer func() {
```

Comment:
this one is tricky: if this is a remote build, and a sandbox is requested as the output, the sandbox is created in a deferred function :sob: